src/main/java/org/opentripplanner/model/plan/StopArrival.java

```diff
@@ -20,11 +20,28 @@
      */
     public final Calendar departure;
 
+    /**
+     * For transit trips, the stop index (numbered from zero from the start of the trip).
+     */
+    public final Integer stopIndex;
```

Comment:
In the transit routing a `stopIndex` mean the global index for each stop, while the stop position along the route is named `stopPosInPattern` - I find it confusing to use the word index here. The name do not explain the context for index, hence it become very easy to misinterpret it. You said a trip may have multiple patterns, but not in this context - a trip on a given date have one pattern, so I think using `stopPosInPattern` is better - at least consistent with all of the code and documantation in the transit routing (internal model of OTP).

Reply:
This is true, I changed it in the other places in #3826, but seems this was not touched. I will change it